frontend/selectors/index.js

```diff
@@ -166,7 +166,7 @@ export const getShopifyVariant = createSelector(
   getProductById,
   (product) => {
     const { productData } = product || {};
-    const { customData = null } = productData || {};
+    const customData = productData.customData || productData.rechargeCustomData;
```

Comment:
In testing I found a problem when I load the product detail page directly. It seems that the PdpRechargeProucts portal can be added to the DOM before the product information is ready.
![image](https://user-images.githubusercontent.com/11764719/67040230-ac947480-f0e8-11e9-93b5-e6c62f37cbc0.png)
I am not sure if it could happen in real life but there is a simple fix. Something like this will fix it. Maybe you can come up with something more elegant.
```
export const getShopifyVariant = createSelector(
  getProductById,
  (product) => {
    const { productData = {} } = product || {};
    const customData = productData.customData || productData.rechargeCustomData;
    if (!customData) {
      return null;
    }
    const { variant_id: shopifyVariantId } = JSON.parse(customData) || {};

    return shopifyVariantId ? `${shopifyVariantId}` : null;
  }
);
```

Reply:
I think that does make sense and clearest solution. Thanks!